sys-fs/mergerfs/mergerfs-2.24.2.ebuild

```diff
@@ -0,0 +1,98 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit toolchain-funcs autotools
+
+DESCRIPTION="A featureful FUSE union filesystem across multiple drives"
+HOMEPAGE="https://github.com/trapexit/mergerfs"
+# tarball under download/ has prebuilt ChangeLog and AUTHORS,
+# and more importantly version.hpp,
+# so use it instead of git tag tarball
+SRC_URI="https://github.com/trapexit/mergerfs/releases/download/${PV}/${P}.tar.gz"
+
+LICENSE="ISC"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="+xattr"
+# mergerfs bundles its own _patched_ copy of libfuse.
+# No easy way around it yet. :(
+#IUSE="${IUSE} +external-fuse"
+
+DEPEND="xattr? ( sys-apps/attr )"
+RDEPEND="${DEPEND}"
+
+PATCHES=(
+	"${FILESDIR}"/${P}-Makefile-don-t-touch-LDFLAGS.patch
+)
+
+DOCS=(
+	AUTHORS
+	ChangeLog
+	README.md
+)
+
+domake() {
+	local makeargs=( "$@" )
+	local makevars=(
+		GIT=''
+		MKDIR="$(type -P mkdir)"
+		TOUCH="$(type -P touch)"
+		CP="$(type -P cp)"
+		RM="$(type -P rm)"
+		LN="$(type -P ln)"
+		INSTALL="$(type -P install)"
+		STRIP='true'
+		PANDOC='' # don't rebuild manpages
+
+		XATTR_AVAILABLE="$(usex xattr 1 0)"
+		PREFIX="${EPREFIX:-/usr}"
+		CXX="$(tc-getCXX)"
+		OPTS="${CPPFLAGS} ${CXXFLAGS}"
+	)
+
+	emake "${makevars[@]}" "${makeargs[@]}"
+}
+
+src_prepare() {
+	default
+
+	# don't rebuild version.hpp
+	cat > tools/update-version <<EOFF
+#!/bin/sh
+exit 0
+EOFF
+
+	pushd libfuse >/dev/null
```

Comment:
I'm confused. Why do all actions except for install need to be done in `libfuse`?

Also, all external commands need `||die`.

Reply:
mergerfs bundles its own copy of (patched) libfuse.
(it does compile against vanilla libfuse-2.7.9, but I don't know the effect of effectively dropping those patches)

Thus, it tries to `autoreconf`, `configure`, and `make` it during `make all`.
First, it somewhat violates `src_prepare()/src_configure()/src_compile()` separation.
But most importantly, it likely ignores all toolchain-related variables, {CXX,LD}FLAGS, make opts, which are normally passed in by Portage.
So, I replaced `autoreconf` with explicit `eautoreconf` in the ebuild, `configure` with `econf`, `make` with `emake` to make sure that everything is sane.

There is nothing related to `libfuse` in `src_install()` because it's compiled as a static library,
and thus linked in statically into the resulting executable.

Reply:
Hmm, but now mergerfs is built in `src_install` instead of `src_compile`, isn't it?

Reply:
No, it's built in the end of `src_compile()` by calling `domake`.